tests/stdlib/talgorithm.nim

```diff
@@ -113,3 +113,82 @@ block:
     doAssert binarySearch(moreData, 6) == -1
     doAssert binarySearch(moreData, 4711) == 4
     doAssert binarySearch(moreData, 4712) == -1
+
+# merge
+block:
+  var x = @[1, 7, 8, 11, 21, 33, 45, 99]
+  var y = @[6, 7, 9, 12, 57, 66]
+
+  let merged = merge(x, y)
+  doAssert merged.isSorted
+  doAssert merged == @[1, 6, 7, 7, 8, 9, 11, 12, 21, 33, 45, 57, 66, 99]
+
+block:
+  var x = @[111, 88, 76, 56, 45, 31, 22, 19, 11, 3]
+  var y = @[99, 85, 83, 82, 69, 64, 48, 42, 33, 31, 26, 13]
+
+  let merged = merge(x, y, SortOrder.Descending)
+  doAssert merged.isSorted(SortOrder.Descending)
+  doAssert merged == @[111, 99, 88, 85, 83, 82, 76, 69, 64, 56, 48, 45, 42, 33, 31, 31, 26, 22, 19, 13, 11, 3]
+  # doAssert merged == @[99, 85, 83, 82, 69, 64, 48, 42, 33, 31, 26, 13, 111, 88, 76, 56, 45, 31, 22, 19, 11, 3]
+
+block:
+  var x: seq[int] = @[]
+  var y = @[1]
+
+  let merged = merge(x, y)
+  doAssert merged.isSorted
+  doAssert merged.isSorted(SortOrder.Descending)
+  doAssert merged == @[1]
+
+block:
+  var x = [1, 3, 5, 5, 7]
+  var y: seq[int] = @[]
+
+  let merged = merge(x, y)
+  doAssert merged.isSorted
+  doAssert merged == @x
+
+block:
+  var x: array[0, int]
+  var y = [1, 4, 6, 7, 9]
+
+  let merged = merge(x, y)
+  doAssert merged.isSorted
+  doAssert merged == @y
+
+block:
+  var x: array[0, int]
+  var y: array[0, int]
+
+  let merged = merge(x, y)
+  doAssert merged.isSorted
+  doAssert merged.len == 0
+
+block:
+  var x: seq[int]
+  var y: seq[int]
+
+  let merged = merge(x, y)
+  doAssert merged.isSorted
+  doAssert merged.len == 0
+
+block:
+  type
+    Record = object
+      id: int
+  
+  proc r(id: int): Record =
+    Record(id: id)
+
+  proc cmp(x, y: Record): int =
```

Comment:
`proc cmp(x, y: Record): int = cmp(x.id, y.id)`

Reply:
It's intended, I want to test that.